build.lua

```diff
@@ -22,6 +22,10 @@ flatten = false
 -- To allow writing
 function docinit_hook()
   mkdir(typesetdir .. "/plots")
+  -- Execute cp() manually to keep the directory structure, otherwise if
+  -- "plots/*.table" is appended to docfiles, ".table" files will be copied
+  -- directly to typesetdir, not typesetdir .. "/plots".
+  cp("plots/*.table", docfiledir, typesetdir .. "/plots")
```

Comment:
To keep treatments unified, do similar special treatment for `"images/*.jpg"`?

Reply:
Alternatively, appending the subdir name to `docfiles` does the trick. @josephwright Is this the expected behavior or a misused "feature"?
```diff
diff --git a/build.lua b/build.lua
index 7bf4f678..9e712ddb 100644
--- a/build.lua
+++ b/build.lua
@@ -12,7 +12,7 @@ docfiledir = "./doc/generic/pgf"
 docfiles =
   {
     "RELEASE_NOTES.md", "description.html", -- Part of the release script
-    "color.cfg", "pgfmanual.cfg", "images/*.jpg", "*.tex" -- Build the PDF
+    "color.cfg", "pgfmanual.cfg", "images", "plots", "*.tex" -- Build the PDF
   }
 tdsroot = "generic"
 typesetfiles = {"pgfmanual.tex"}
```

Reply:
@josephwright ping

Reply:
Another idea: if `\input@path`/`\l_file_search_path_seq` is respected by `\pgfutil@IfFileExists` and `\pgfutil@InputIfFileExists` (see #1259), then we can drop prefix `plots/` in manual examples
https://github.com/pgf-tikz/pgf/blob/14ef7f3494766f3df4c9b177bec1e5695ec980f5/doc/generic/pgf/pgfmanual-en-tikz-plots.tex#L146

Reply:
> Alternatively, appending the subdir name to `docfiles` does the trick. @josephwright Is this the expected behavior or a misused "feature"?

From conversation in https://github.com/latex3/l3build/issues/331 and a casual check in `l3build` source, yes it is supported. I tend to use this currently undocumented feature in pgf's `l3build` configuration.

As this PR also promotes missing plot data file from warning to error, feature breaking change in `l3build` will be caught.